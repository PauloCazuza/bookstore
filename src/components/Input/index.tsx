import { Input as NBInput, IInputProps, FormControl } from "native-base";

type Props = IInputProps & {
    errorMessage?: string | null;
}

export default function Input({ errorMessage, isInvalid, ...rest }: Props) {
    const invalid = !!errorMessage || isInvalid;

    return (
        <FormControl mb={4} isInvalid={invalid} >
            <NBInput
                bg="gray.200"
                fontSize="md"
                h={16}
                mb={4}
                _focus={{
                    bg: "gray.300",
                    borderWidth: 2,
                    borderColor: "green.500"
                }}
                _invalid={{
                    borderWidth: 1,
                    borderColor: "red.500"
                }}
                {...rest}
            />

            <FormControl.ErrorMessage>
                {errorMessage}
            </FormControl.ErrorMessage>

        </FormControl>
    )
}