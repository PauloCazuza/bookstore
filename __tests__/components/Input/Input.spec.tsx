import { NativeBaseProvider } from "native-base";
import renderer from "react-test-renderer";
import Input from "../../../src/components/Input/index";

it('renders correctly Input', () => {
    const tree = renderer.create(<NativeBaseProvider><Input value="teste" /></NativeBaseProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});