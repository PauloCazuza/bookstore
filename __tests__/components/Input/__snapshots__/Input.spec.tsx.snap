// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Input 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
